join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
 
-```graphql definition
-enum join__Graph
+In order for the router to send an operation that resolves a given field on a parent object to a subgraph, the operation needs to first resolve the parent object itself. There are several ways to accomplish this, described below. The examples below include abbreviated versions of the supergraph schemas which do not include the `schema` definition, directive definitions, or the `join__Graph` definition. This specification does not require the subgraph queries to be the same as those described in these examples; this is just intended to broadly describe the meanings of the directives.
+
+## Root fields
+
+If this field appears at the root of the overall operation (query or mutation), then the field can just be placed at the root of the subgraph operation.
+
+```graphql example -- Root fields
+# Supergraph schema
+type Query {
+  fieldA: String @join__field(graph: A)
+  fieldB: String @join__field(graph: B)
+}
+
+# Operation
+{ fieldA fieldB }
+# Generated subgraph operations
+## On A:
+{ fieldA }
+## On B:
+{ fieldB }
 ```
 
-Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
-:::[example](photos.graphql#join__Graph) -- Using join__Graph to define endpoints
+## Fields on the same subgraph as the parent operation
 
-The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
+If this field's parent field will be resolved by an operation on the same subgraph, then this field can be resolved as part of the same operation, by putting it in a nested selection set on the parent field's subgraph operation. Note that this example contains a {@join__owner} directive on an object type; this will be described later.
 
-# Directives
+```graphql example -- Fields on the same subgraph as the parent operation
+# Supergraph schema
+type Query {
+  fieldA: X @join__field(graph: A)
+}
 
-##! @join__graph
+type X @join__owner(graph: A) {
+  nestedFieldA: String @join__field(graph: A)
+}
 
-Declare subgraph metadata on {join__Graph} enum values. 
+# Operation
+{ fieldA { nestedFieldA } }
+# Generated subgraph operations
+## On A:
+{ fieldA { nestedFieldA }}
+```
 
-```graphql definition
-directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+## Fields provided by the parent field
+
+When resolving a field to a composite type (object, interface, or union), it is sometimes easy to calculate the values of nested fields inside that object even if those nested fields are ordinarily resolved in a different subgraph. When this is the case, you can include a `provides` argument in the `@join__field` listing these "pre-calculated" fields. The router can now resolve these fields in the "providing" subgraph instead of in the subgraph that would usually be used to resolve those fields.
```

Comment:
```suggestion
Sometimes, a subgraph may be capable of resolving nested fields (which it does not contribute to the graph) on a type. When this is the case, you can include a `provides` argument in the `@join__field` listing these "pre-calculated" fields. The router can now resolve these fields in the "providing" subgraph instead of in the subgraph that would usually be used to resolve those fields.
```

I honestly don't know how to reword this but this is where I'm at. I think the second half of what you have reads very well. The first is a bit confusing (and clearly difficult to articulate well, since I'm struggling to come up with something that I'm happy with).

Reply:
Hmm. I agree that my wording was confusing but I am not sure about your one either. Maybe an example would help? d010a05 tries a different approach for the first sentence plus a more motivating example.

Reply:
This is a huge improvement! Really like how the explanation turned out (and the examples are great too)